core/src/impl/Kokkos_SharedAlloc.cpp

```diff
@@ -285,22 +286,28 @@ void SharedAllocationRecord<void, void>::print_host_accessible_records(
   // allocation.
   const SharedAllocationRecord<void, void>* r = root->m_next;
 
+#define KOKKOS_PAD_HEX(ptr)                              \
+  "0x" << std::hex << std::setw(12) << std::setfill('0') \
+       << reinterpret_cast<uintptr_t>(ptr)
```

Comment:
The issue with the current implementation is that the `setfill`, `hex` and `dec` manipulators are sticky.
So if someone prints the record on `std::cout` then subsequent integers that would be put into it will be written has hexes.

Reply:
My bad; fixed!